restclients_core/dao.py

```diff
@@ -239,18 +244,39 @@ def _getModule(self, value, default_class, args=[]):
         return config_module(*args)
 
     def _log(self, *args, **kwargs):
-        log_timing = self.get_setting("TIMING_LOG_ENABLED", False)
-        logging_rate = float(self.get_setting("TIMING_LOG_RATE", 1.0))
-
-        if log_timing and random.random() <= logging_rate:
-            from_cache = 'yes' if kwargs.get('cached') else 'no'
-            total_time = time.time() - kwargs.get('start_time')
-            msg = (('service:%s method:%s url:%s status:%s from_cache:%s' +
-                   ' time:%s')
-                   % (kwargs.get('service'), kwargs.get('method'),
-                      kwargs.get('url'), kwargs.get('status'),
-                      from_cache, total_time))
-            logger.info(msg)
+        if not self.should_log():
+            return
+
+        from_cache = 'yes' if kwargs.get('cached') else 'no'
+        total_time = time.time() - kwargs.get('start_time')
+        msg = (('service:%s method:%s url:%s status:%s from_cache:%s' +
+               ' time:%s')
+               % (kwargs.get('service'), kwargs.get('method'),
+                  kwargs.get('url'), kwargs.get('status'),
+                  from_cache, total_time))
+        logger.info(msg)
+
+    def should_log(self):
+        log_timing = self.get_service_setting("TIMING_LOG_ENABLED", False)
+        logging_rate = float(self.get_service_setting("TIMING_LOG_RATE", 1.0))
+
+        # format is ISO 8601
+        log_start_str = self.get_service_setting("TIMING_START", None)
+        log_end_str = self.get_service_setting("TIMING_END", None)
```

Comment:
would it be better to move the above 4 lines into the init method so that they are set only once?